pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -93,124 +117,172 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	}
 	labels["app"] = el.Name
 
-	// Create the EventListener Deployment
-	deployment := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			// Create the EventListener's Deployment in the same Namespace as where
-			// the EventListener was created
-			Namespace: el.Namespace,
-			// Give the Deployment the same name as the EventListener
-			Name: el.Name,
-			// If our EventListener is deleted, then its Deployment should be as well
-			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
-			Labels:          labels,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Selector: &metav1.LabelSelector{
+	oldService, err := c.KubeClientSet.CoreV1().Services(el.Namespace).Get(el.Name, metav1.GetOptions{})
+	switch {
```

Comment:
Can we have two functions, say reconcileService and reconcileDeployment that take in the old and new resources? Will make it easy to write targeted unit tests in addition to the reconciler tests